timemachine/fe/single_topology.py

```diff
@@ -1615,6 +1471,219 @@ def _setup_intermediate_chiral_bond_term(
+
+    def _angle_idxs_belong_to_chiral_volume_turning_off(self, idxs):
+        src_chiral_idxs = set(tuple(x) for x in self.src_system.chiral_atom.potential.idxs)
+        dst_chiral_idxs = set(tuple(x) for x in self.dst_system.chiral_atom.potential.idxs)
+        induced_angle_idxs = set()
+        for c, i, j, k in src_chiral_idxs.difference(dst_chiral_idxs):
+            induced_angle_idxs.add(canonicalize_bond((i, c, j)))
+            induced_angle_idxs.add(canonicalize_bond((i, c, k)))
+            induced_angle_idxs.add(canonicalize_bond((j, c, k)))
+
+        return idxs in induced_angle_idxs
+
+    def _interpolate_bond(self, idxs, src_params, dst_params, lamb):
+        if self.all_idxs_belong_to_core(idxs):
+            min_max = CORE_BOND_MIN_MAX
+        elif self.any_idxs_belong_to_dummy_a(idxs):
+            if self._bond_idxs_belong_to_chiral_volume_turning_on(idxs):
+                assert 0
+            elif self._bond_idxs_belong_to_chiral_volume_turning_off(idxs):
+                min_max = DUMMY_A_CHIRAL_BOND_CONVERTING_OFF_MIN_MAX
+            else:
+                min_max = DUMMY_A_BOND_MIN_MAX
+        elif self.any_idxs_belong_to_dummy_b(idxs):
+            if self._bond_idxs_belong_to_chiral_volume_turning_on(idxs):
+                min_max = DUMMY_B_CHIRAL_BOND_CONVERTING_ON_MIN_MAX
+            elif self._bond_idxs_belong_to_chiral_volume_turning_off(idxs):
+                assert 0
+            else:
+                min_max = DUMMY_B_BOND_MIN_MAX
+        else:
+            assert 0
+
+        k_min = 0.025
+        return interpolate_harmonic_bond_params(src_params, dst_params, lamb, k_min, *min_max)
+
+    def _interpolate_angle(self, idxs, src_params, dst_params, lamb):
+        k_min = 0.05
+        if self.all_idxs_belong_to_core(idxs):
+            if self._angle_idxs_belong_to_chiral_volume_turning_on(idxs):
+                min_max = CORE_CHIRAL_ANGLE_CONVERTING_ON_MIN_MAX
+            elif self._angle_idxs_belong_to_chiral_volume_turning_off(idxs):
+                min_max = CORE_CHIRAL_ANGLE_CONVERTING_OFF_MIN_MAX
+            else:
+                min_max = CORE_ANGLE_MIN_MAX
+        elif self.any_idxs_belong_to_dummy_a(idxs):
+            if self._angle_idxs_belong_to_chiral_volume_turning_on(idxs):
+                assert 0
+            elif self._angle_idxs_belong_to_chiral_volume_turning_off(idxs):
+                min_max = DUMMY_A_CHIRAL_ANGLE_CONVERTING_OFF_MIN_MAX
+            else:
+                min_max = DUMMY_A_ANGLE_MIN_MAX
+        elif self.any_idxs_belong_to_dummy_b(idxs):
+            if self._angle_idxs_belong_to_chiral_volume_turning_on(idxs):
+                min_max = DUMMY_B_CHIRAL_ANGLE_CONVERTING_ON_MIN_MAX
+            elif self._angle_idxs_belong_to_chiral_volume_turning_off(idxs):
+                assert 0
+            else:
+                min_max = DUMMY_B_ANGLE_MIN_MAX
+        else:
+            assert 0
+
+        k_min = 0.05
```

Comment:
```suggestion
```
(already defined a few lines earlier in this function)

Reply:
willfix